### PR TITLE
feat: bring transaction response type into shared package

### LINKED ISSUE #1338: [Subtask] - Share types for transaction
Share types for Testnet Wallet backend and Testnet Boutique backend
Controllers:

transaction

### PR DESCRIPTION
**Pull request by rico191013**
### Context

- fixes #1338

### CODE REVIEW

**Review by beniaminmunteanu — APPROVED** [review]
LGTM
Mentioning that this approve is provided without starting the local dev in order to see the transactions list displayed in the frontend

File: packages/wallet/shared/src/types/transaction.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+export type TransactionType = keyof typeof TRANSACTION_TYPE
+
+const TRANSACTION_STATUS = {
+  PENDING: 'PENDING',
+  COMPLETED: 'COMPLETED',
+  REJECTED: 'REJECTED'
+} as const
+type TransactionStatus = keyof typeof TRANSACTION_STATUS
+
+export interface TransactionResponse {
```
Comment (dragosp1011) [inline packages/wallet/shared/src/types/transaction.ts:14]: try to make the transaction model to implement this interface

File: packages/wallet/shared/src/types/transaction.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+const TRANSACTION_TYPE = {
```
Comment (dragosp1011) [inline packages/wallet/shared/src/types/transaction.ts:1]: can these be enums?

File: packages/wallet/shared/src/types/transaction.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  EXPIRED: 'EXPIRED',
+  FAILED: 'FAILED'
+} as const
+type TransactionStatus = keyof typeof TRANSACTION_STATUS
+
+export interface TransactionResponse {
+  id: string
+  paymentId: string
+  walletAddressPublicName?: string
+  walletAddressUrl?: string
```
Comment (dragosp1011) [inline packages/wallet/shared/src/types/transaction.ts:19]: should be part of `TransactionListResponse`

File: packages/wallet/shared/src/types/transaction.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  description?: string
+  walletAddressId?: string
```
Comment (dragosp1011) [inline packages/wallet/shared/src/types/transaction.ts:21]: should be part of `TransactionListResponse`

File: packages/wallet/shared/src/types/transaction.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  assetCode: string
+  value: string | null | bigint
+  type: TransactionType
+  status: TransactionStatus
+}
+interface TransactionListResponse extends TransactionResponse {
+  assetScale: number
+  accountName: string
+  createdAt: string
```
Comment (dragosp1011) [inline packages/wallet/shared/src/types/transaction.ts:30]: `createdAt`  and `updatedAt` should be part of `TransactionResponse`

File: packages/wallet/shared/src/types/transaction.ts (hunk 2 above)
Comment (dragosp1011) [reply packages/wallet/shared/src/types/transaction.ts:1]: @rico191013 can this be transformed to enum as well?

File: packages/wallet/frontend/src/pages/transactions.tsx [hunk 6]
```diff
@@ -323,7 +323,7 @@ const TransactionsPage: NextPageWithLayout<TransactionsPageProps> = ({
                       />
                     </Table.Cell>
                     <Table.Cell className="whitespace-nowrap">
-                      {formatDate({ date: trx.createdAt })}
+                      {formatDate({ date: trx.createdAt.toString() })}
```
Comment (dragosp1011) [inline packages/wallet/frontend/src/pages/transactions.tsx:326]: isn't it string already?
https://github.com/interledger/testnet/blob/main/packages/shared/backend/src/model.ts#L31

**Review by dragosp1011 — APPROVED** [review]

**Pull request merged**